io/src/test/scala/fs2/io/tcp/SocketSpec.scala

```diff
@@ -127,5 +128,44 @@ class SocketSpec extends Fs2Spec {
           .get
       result shouldBe sizes
     }
+
+    "write - concurrent calls do not cause WritePendingException" in {
+      val message = Chunk.bytes(("123456789012345678901234567890" * 10000).getBytes)
+
+      val localBindAddress =
+        Deferred[IO, InetSocketAddress].unsafeRunSync()
+
+      def doNothingServer(socketGroup: SocketGroup) =
+        socketGroup
+          .serverResource[IO](new InetSocketAddress(InetAddress.getByName(null), 0))
+          .flatMap {
+            case (address, _) => Resource.liftF(localBindAddress.complete(address))
+          }
+
+      val result =
+        mkSocketGroup
+          .flatMap { socketGroup =>
+            Stream.resource(doNothingServer(socketGroup)) >>
+              Stream.eval(localBindAddress.get).flatMap { address =>
+                Stream
+                  .resource(
+                    socketGroup.client[IO](address)
+                  )
+                  .flatMap(sock =>
+                    Stream(
+                      Stream.eval(sock.write(message)).repeatN(10L)
+                    ).repeatN(2L)
+                  )
+                  .parJoinUnbounded
+              }
+          }
+          .compile
+          .drain
+          .attempt
+          .unsafeRunTimed(timeout)
```

Comment:
I'd suggest using `timeout,` instead. `unsafezrunTimed` limits the time between async boundaries, not the whole IO.